### ISSUE TITLE
Implement log(cosh()) loss function

### ISSUE THREAD

**Issue by n2cholas** [deepmind/optax#103]
log-cosh is a doubly differentiable alternative to the huber loss. A naive implementation is prone to overflow (since cosh has an e^x term), so I think it'd be a useful addition to the library. Plus, it's implemented in other libraries, such as [TensorFlow](https://www.tensorflow.org/api_docs/python/tf/keras/losses/log_cosh).

If this sounds like a relevant addition, I'd be happy to contribute it!

**Comment by mtthss** [comment]
This would be a great contribution! Looking forward to it!
